test/config-luatest/jsonschema_config_test.lua

```diff
@@ -0,0 +1,103 @@
+local cluster_config = require("internal.config.cluster_config")
+local config = require('config')
+local t = require('luatest')
+
+local g = t.group()
+
+local function remove_descriptions(tbl)
+    if type(tbl) ~= 'table' then
+        return
+    end
+
+    for key, value in pairs(tbl) do
+        if key == 'description' then
+            tbl[key] = nil
+        elseif type(value) == 'table' then
+            remove_descriptions(value)
+        end
+    end
+end
+
+g.test_cluster_config_schema_description_completeness = function()
+    local function check_schema_description(schema, ctx)
+        local field_path = table.concat(ctx.path, '.')
+        t.assert(schema.description ~= nil or field_path == '',
+                 string.format("%q is missing description", field_path))
+        if schema.type == 'record' then
+            for field_name, field_def in pairs(schema.fields) do
+                table.insert(ctx.path, field_name)
+                check_schema_description(field_def, ctx)
+                table.remove(ctx.path)
+            end
+        elseif schema.type == 'map' then
+            table.insert(ctx.path, '*')
+            check_schema_description(schema.value, ctx)
+            table.remove(ctx.path)
+        elseif schema.type == 'array' then
+            table.insert(ctx.path, '*')
+            check_schema_description(schema.items, ctx)
+            table.remove(ctx.path)
+        end
+    end
+
+    local cluster_config_schema = rawget(cluster_config, 'schema')
+    t.assert(cluster_config_schema ~= nil)
+    check_schema_description(cluster_config_schema.fields, {path = {}})
```

Comment:
The commit is about config's JSON Schema generation, the test file is about it as well. Maybe verify `config:jsonschema()` instead of the internal cluster config module?

Alternatively we can move this test case to anothre file (not related to JSON Schemas) and move it to the commit, where the descriptions are added.

Reply:
I replaced this test with a test with `config:jsonschema()`. The old test was moved to tests where descriptions were added.